src/framework/uima/flow.hpp

```diff
@@ -0,0 +1,207 @@
+#ifndef UIMA_FLOW_HPP
+#define UIMA_FLOW_HPP
+
+/** \file flow.hpp .
+-----------------------------------------------------------------------------
+
+
+
+
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+
+-----------------------------------------------------------------------------
+
+   Description: Flow interface for the FlowController
+
+-----------------------------------------------------------------------------
+
+
+-------------------------------------------------------------------------- */
+#include <memory>
+
+#include "uima/pragmas.hpp"
+#include "uima/annotator.hpp"
+
+namespace uima {
+  namespace internal {
+    /**
+     * Indicates that a CAS should be routed to a single AnalysisEngine.
+     */
+    class UIMA_LINK_IMPORTSPEC SimpleStep {
+      /* The key of the engine the CAS will be input to.
+       Not to be confused with the engine name, which is specified by the its descriptor*/
+      icu::UnicodeString engineKey;
+      /* ResultSpecification *resultSpec; */
+    public:
+      /* If SimpleStepWithResultSpec is required when CapabilityLanguageFlowController is implemented,
+       * this class could have an extra member ResultSpecification
+      SimpleStep(const icu::UnicodeString &name, ResultSpecification *resultSpec) : engineKey(name) {
+      } */
+
+      SimpleStep(const icu::UnicodeString &name) : engineKey(name) {
+      }
+
+      const icu::UnicodeString &getEngineName() const {
+        return engineKey;
+      }
+
+      /**
```

Comment:
why this code is commented out? Either add it or delete it.

Reply:
There is a separate type in the Java version that's SimpleStepWithResultSpec which is just used for the Capability Language Flow. Since we're not adding it and using the old code right now, I wanted to leave it there in case we want to implement that in the future.